### ISSUE TITLE
Please, please, please implement Telegram AD support in order to prevent app from ban

### ISSUE THREAD

**Issue by wvolkov** [hihain/Unigram#41]
It is necessary to implement support of telegram ad feature, otherwise Telegram API will be blocked for third-party clients, thus Windows Mobile 10 Users will be abandoned from the last working messenger. Praying for your support 🙏🙏🙏🙏🙏

**Comment by romanicak** [comment]
Yes, there are many of us using Telegram (which is not able to login from 03.12.2021) or will be using Unigram mobile from now on Windows 10 mobile devices.

**Comment by wvolkov** [comment]
Trying to install woa at Lumia 950 xl as workaround

**Comment by wonderkast02** [comment]
I also ask please to update, I also use it frequently on my 640 XL, pray for your support.

**Comment by AnonymousWebHacker** [comment]
i FOUND https://4pda.to/forum/index.php?showtopic=990472&view=findpost&p=111436601

could you verify if it works?

**Comment by wvolkov** [comment]
It seems a PWA which works like a browser wrapper under the hood with the link to legacy telegram web interface https://web.telegram.org/?legacy=1.I have checked it at my Lumia 950 Xl - it seems that its edge browser core is so obsolete that it didn't work: couldn't move to the next step. You can try it your own by installing this appx bundle (make sure to check a "developer" mode at update settings, download file below and rename its extension from mp4 to appx

https://user-images.githubusercontent.com/49549154/144736980-0da984ab-5dc5-4ce0-9124-5148fec5103e.mp4

**Comment by wvolkov** [comment]
A bit offtop. As a workaround, Lumia 950XL users may try to install a Windows 10 on arm build. I tried it yesterday and it works quite handy everything works rather smooth, of course there is some lags but it is something you could live. 
Pros:
- arm apps for elder brother windows 10 work quite good: Edge Chromium Browser / Telegram / Etc.
- support for phone calls
- x86 apps emulation (but too slow)
- dual boot (select windows phone or 10 at startup)

Cons:
- no camera support
- battery drain much more than usual windows phone os
- only one Sim support in case of dual sim devices
- risk to brick your device while installing (need to carefully read the instructions)



<details>
<summary>Screenshots attached</summary>
<img src="https://user-images.githubusercontent.com/49549154/144737025-01b28771-fd72-411e-8e4a-152ffa5285d4.jpg" width=50%/>
<img src="https://user-images.githubusercontent.com/49549154/144737055-c05141df-0c53-421c-8932-3db52803db61.jpg"  width=50%/>
<img src="https://user-images.githubusercontent.com/49549154/144737056-be31047e-8803-4afd-9b5d-f0c856a54472.jpg" width=50%/>
<img src="https://user-images.githubusercontent.com/49549154/144737057-95f059ee-39c1-4be2-b7ce-8921045b5718.jpg" width=50%/>
<img src="https://user-images.githubusercontent.com/49549154/144737058-dc5caff5-0dc8-4b58-9e53-2ea21a8d7487.jpg" width=50%/>
</details>

**Comment by AnonymousWebHacker** [comment]
I tried the app PWA it and it did not work for me, I tried the app and the edge and the same result, the key is generated, but it does not happen there. I imagine it is because of the outdated web-tools

**Comment by amigalover** [comment]
I tried the legacy web app, too. Same problem, it doesn't pass the next step after generating the keys.
So on Lumia 640 / 650 the app does not work nor the Edge browser :-(
Hopefully the coder of Unigram is/are possible to get it to work again. Good luck!

**Comment by dirijablik** [comment]
One hope. Please... This is really very important for many

**Comment by ahorn42** [comment]
> It is necessary to implement support of telegram ad feature, otherwise Telegram API will be blocked for third-party clients, thus Windows Mobile 10 Users will be abandoned from the last working messenger. Praying for your support 🙏🙏🙏🙏🙏

do you have link or message or some how more information what would be necessary to do?

I also really hope this can be fixed!

**Comment by ahorn42** [comment]
The "parent" of this repository (https://github.com/UnigramDev/Unigram) seems to still work.

**Comment by wvolkov** [comment]
> > It is necessary to implement support of telegram ad feature, otherwise Telegram API will be blocked for third-party clients, thus Windows Mobile 10 Users will be abandoned from the last working messenger. Praying for your support 🙏🙏🙏🙏🙏
> 
> do you have link or message or some how more information what would be necessary to do?
> 
> I also really hope this can be fixed!

Here is the link https://core.telegram.org/api/sponsored-messages#sponsored-messages-in-third-party-apps

Below is an answer from official developer at his telegram channel:

> I am preparing an update, that reflects these changes. It will be released ASAP. Certainly this month.
Please try to be patient. I already committed 189 times many changes. Both release and beta version will be released ASAP. It depends on how fast I can fix the incompatibilities to the new TDlib version and squash bugs preventing Unigram Mobile to be usable. As a result: Do not expect Unigram Mobile to be stable. It might crash. Currently, everything is better that not being able to use Telegram on Windows Mobile. Not even the old official clients work anymore....

So let's be patient and wait for release and not interrupt him like this donkey from Shrek 2 cartoon 😁

**Comment by ahorn42** [comment]
That sounds great - in terms of he is allready on it ;)

Hope he gets it running soon - is there a place we can leave him a tip for his work? :)

**Comment by hihain** [comment]
The good news: I pushed an update for the beta testers to the Store.
The bad news: It was impossible to debug / test the app. Therefore the beta release has to show me (and others) that it starts.

So let's wait for the Microsoft certification process to finish to check if the beta works well enough to be asap released to the general public.

AD support will be added at a later time.

Sorry for not answering here. I was busy with the update.

**Comment by mediaexplorer74** [comment]
> The "parent" of this repository (https://github.com/UnigramDev/Unigram) seems to still work.

Does Unigram ("Parent project" as you said) supports Win10Mobile?... I think its for Win10/11 Desktop only (1 year or more)

**Comment by ahorn42** [comment]
> > The "parent" of this repository (https://github.com/UnigramDev/Unigram) seems to still work.
> 
> Does Unigram ("Parent project" as you said) supports Win10Mobile?... I think its for Win10/11 Desktop only (1 year or more)

yes, that's right, I only wanted to point out, that it is not a general 3rd-party app problem and that there is a very similar project were we might be able to look for an easy to translate solution ;)

beside this, I am very happy to see that there are still some windows mobile users left :) - too bad our platform has been deprecated, with devices that are still quite capable as daily driver. I wish they had pushed oen further release to the lumias, so we could use dot net standard 2.0, this would open up for so much more applications and libraries :/
